SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h

```diff
@@ -6,11 +6,49 @@
 #include "Schema/UnrealObjectRef.h"
 #include "SpatialConstants.h"
 
-#include "CoreMinimal.h"
+#include "Core.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogRPCContainer, Log, All);
 
 struct FPendingRPCParams;
+struct FRPCErrorInfo;
 using FPendingRPCParamsPtr = TUniquePtr<FPendingRPCParams>;
-DECLARE_DELEGATE_RetVal_OneParam(bool, FProcessRPCDelegate, const FPendingRPCParams&)
+DECLARE_DELEGATE_RetVal_OneParam(FRPCErrorInfo, FProcessRPCDelegate, const FPendingRPCParams&)
+
+enum class ERPCError : uint8_t
+{
+	Success,
+
+	// Shared across Sender and Receiver
+	UnresolvedTargetObject,
+	MissingFunctionInfo,
+	UnresolvedParameters,
+
+	// Sender specific
+	NoActorChannel,
+	SpatialActorChannelNotListening,
+	NoNetConnection,
+	NoAuthority,
+	InvalidRPCType,
+
+	// Specific to packing
+	NoOwningController,
+	NoControllerChannel,
+	UnresolvedController,
+	ControllerChannelNotListening
+};
+
+struct FRPCErrorInfo
+{
+	bool Success() const
+	{
+		return (ErrorCode == ERPCError::Success);
+	}
+
+	UObject* TargetObject;
```

Comment:
I think it makes sense to use a TWeakObjectPtr<> here.

Reply:
Updated.